clab/exec/exec.go

```diff
@@ -51,22 +51,27 @@ func NewExecCmdFromSlice(cmd []string) *ExecCmd {
 	}
 }
 
-type ExecResultHolder interface {
-	GetStdOutString() string
-	GetStdErrString() string
-	GetStdOutByteSlice() []byte
-	GetStdErrByteSlice() []byte
-	GetReturnCode() int
-	GetCmdString() string
-	Dump(format string) (string, error)
-	String() string
+// Stdout type alias for a string is an artificial type
+// to allow for custom marshalling of stdout output which can be either
+// a valid or non valid JSON.
+// For that reason a custom MarshalJSON method is implemented to take care of both.
+type Stdout string
+
+// MarshalJSON implements a custom marshaller for a custom Stdout type.
+func (s Stdout) MarshalJSON() ([]byte, error) {
+	switch {
+	case json.Valid([]byte(s)):
+		return []byte(s), nil
+	default:
+		return json.Marshal(string(s))
+	}
 }
 
 // ExecResult represents a result of a command execution.
 type ExecResult struct {
 	Cmd        []string `json:"cmd"`
 	ReturnCode int      `json:"return-code"`
-	Stdout     string   `json:"stdout"`
+	Stdout     Stdout   `json:"stdout"`
 	Stderr     string   `json:"stderr"`
 }
```

Comment:
@steiler the idea here is to use an alias type and implement a custom marshaller for it.
if the stdout contains a valid json the marshaller just return those bytes, as they are already in shape

else, it uses the stdlib json.Marshal over the string type